frontend/src/components/Sidebar/Sidebar.tsx

```diff
@@ -112,7 +112,7 @@ function SidebarToggleButton() {
   );
 }
 
-function DefaultLinkArea(props: { sidebarName: string; isOpen: boolean }) {
+const DefaultLinkArea = memo((props: { sidebarName: string; isOpen: boolean }) => {
```

Comment:
Is this really needed? I thought React already only rendered components only if the props change, and there's no complex work in the body of the component AFAICS.

Reply:
React rerenders component if the parent updates, even if the props are the same. https://react.dev/reference/react/memo#skipping-re-rendering-when-props-are-unchanged

Since the props here are super simple, comparing if props are changed is not an expensive operation